gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/JDBCTokenStateService.java

```diff
@@ -193,8 +194,17 @@ protected void evictExpiredTokens() {
 
   @Override
   public void addMetadata(String tokenId, TokenMetadata metadata) {
+
     try {
-      final boolean added = tokenDatabase.addMetadata(tokenId, metadata);
+      boolean added = false;
```

Comment:
Chance of running into concurrency issue here with the flag.

Reply:
Thanks, @moresandeep ! I've just submitted a new PS with the fix.